src/components/calcite-button/calcite-button.scss

```diff
@@ -114,6 +114,72 @@
   width: 100%;
 }
 
+// alignment
+:host([alignment="center"]:not([width="auto"])) {
+  a,
+  button {
+    justify-content: center;
+  }
+}
+:host([alignment="start"]:not([width="auto"])) {
+  a,
+  button {
+    justify-content: flex-start;
+  }
+}
+:host([alignment="end"]:not([width="auto"])) {
+  a,
+  button {
+    justify-content: flex-end;
+  }
+}
+:host([alignment*="space-between"]:not([width="auto"])) {
+  a,
+  button {
+    justify-content: space-between;
+  }
+}
+:host([alignment="space-around"]:not([width="auto"])) {
+  a,
+  button {
+    justify-content: space-around;
+  }
+}
+:host([icon-start][alignment="icon-start-space-between"]:not([width="auto"])) {
+  a,
+  button {
+    .icon-start {
+      margin-right: auto; // see RTL
+    }
+  }
+}
+:host([icon-end][alignment="icon-end-space-between"]:not([width="auto"])) {
+  a,
+  button {
+    .icon-end {
+      margin-left: auto; // see RTL
+    }
+  }
+}
+:host([dir="rtl"][icon-start][alignment="icon-start-space-between"]:not([width="auto"])) {
```

Comment:
@macandcheese @paulcpederson Do you suggest a more succinct way to do this?

Reply:
Yeah we really need to not make this so complicated in CSS 👍 

@asangma can you just add a class to the node you need instead when the properties are set to what you expect?

Reply:
Yeah I'd probably just add another class to "calcite-button--icon icon-start/end" if that appearance type is requested. 

Or with just css you could drop the nested "a, button", and the [icon-start], and just apply to the child, since an .icon-start will only be there if the attribute is there anyway.

:host([dir="rtl"][alignment="icon-start-space-between"]:not([width="auto"])) .calcite-button--icon.icon-start {
}

Reply:
Thanks! I was hoping to do that but wasn't sure if there was a good reason to use the property in the selector. Will update.

Reply:
@macandcheese @driskull Updated.

Reply:
Also, the `:not([width="auto"])` is accurate and specific, but I wonder if it's really necessary.

Reply:
Hm, not sure if having those properties in an auto width button would cause issues in flex layouts? Maybe? But does seem like we shouldn't set those properties if it's width auto to prevent anything like that from happening.